autotest/gdrivers/ogcapi.py

```diff
@@ -0,0 +1,185 @@
+                        + self.path.replace("/fakeogcapi", ""),
+                        stream=True,
+                    )
+                    content = response.content.replace(
+                        b"https://demo.pygeoapi.io/stable",
+                        (
+                            "http://"
+                            + self.address_string()
+                            + ":"
+                            + str(self.server.server_port)
+                            + "/fakeogcapi"
+                        ).encode("utf8"),
+                    )
+                    data = b"HTTP/2 200 OK\r\n"
+                    for k, v in response.headers.items():
+                        if k == "Content-Length":
+                            data += (
+                                k.encode("utf8")
+                                + b": "
+                                + str(len(content)).encode("utf8")
+                                + b"\r\n"
+                            )
+                        else:
+                            data += (
+                                k.encode("utf8") + b": " + v.encode("utf8") + b"\r\n"
+                            )
+                    data += b"\r\n"
+                    data += content
+                    fd.write(data)
+                    self.wfile.write(data)
+
+            elif self.path.find("/fakeogcapi") != -1:
+
+                with open(request_data_path, "rb+") as fd:
+                    self.wfile.write(fd.read())
+
+            return
+
+        except IOError:
+            pass
+
+        self.send_error(404, "File Not Found: %s" % self.path)
+
+
+###############################################################################
+# Test underlying OGR drivers
+#
+
+
+pytestmark = pytest.mark.require_driver("OGCAPI")
+
+###############################################################################
+@pytest.fixture(autouse=True, scope="module")
+def module_disable_exceptions():
+    with gdaltest.disable_exceptions():
+        yield
+
+
+@pytest.fixture(autouse=True, scope="module")
+def ogr_ogcapi_init():
+    gdaltest.ogcapi_drv = ogr.GetDriverByName("OGCAPI")
+
+
+def test_ogr_ogcapi_fake_ogcapi_server():
+    if gdaltest.ogcapi_drv is None:
+        pytest.skip()
+
+    (process, port) = webserver.launch(handler=OGCAPIHTTPHandler)
+    if port == 0:
+        pytest.skip()
+
+    ds = gdal.OpenEx("OGCAPI:http://127.0.0.1:%d/fakeogcapi" % port)
+    if ds is None:
+        webserver.server_stop(process, port)
+        pytest.fail("did not manage to open OGCAPI datastore")
+
+    sub_ds_uri = [
+        v[0] for v in ds.GetSubDatasets() if v[1] == "Collection Large Lakes"
+    ][0]
+    del ds
+
+    ds = gdal.OpenEx(sub_ds_uri)
+    assert ds is not None
+
+    lyr = ds.GetLayerByName("lakes")
+
+    if lyr.GetName() != "lakes":
+        print(lyr.GetName())
+        webserver.server_stop(process, port)
+        pytest.fail("did not get expected layer name")
```

Comment:
if that fails, pytest will magically expand the value of lyr.GetName()

```suggestion
    assert lyr.GetName() == "lakes"
```